Lib/csv.py

```diff
@@ -413,27 +414,45 @@ def has_header(self, sample):
             if len(row) != columns:
                 continue # skip rows that have irregular number of columns
 
+            #checking if all col are strings
+            for cols in list(columnTypes.keys()):
+                if row[cols].isnumeric():
+                    col_are_strings = False
+                    break
+
             for col in list(columnTypes.keys()):
                 thisType = complex
+                
+                
                 try:
                     thisType(row[col])
+                    
                 except (ValueError, OverflowError):
                     # fallback to length of string
                     thisType = len(row[col])
-
+                   
+                
                 if thisType != columnTypes[col]:
                     if columnTypes[col] is None: # add new column type
                         columnTypes[col] = thisType
+                        average_size += len(row[col])
                     else:
                         # type is inconsistent, remove column from
                         # consideration
                         del columnTypes[col]
+            
+
+                     
 
         # finally, compare results against first row and "vote"
         # on whether it's a header
         hasHeader = 0
+        # here we added the special case where all cols are strings and dictionnary has been emptied
+        if not columnTypes and col_are_strings==True and columns>0:
+            #dictionary now takes the average length of strings
+            columnTypes[0] = int(average_size/columns)
         for col, colType in columnTypes.items():
-            if isinstance(colType, int): # it's a length
+            if type(colType) == type(0): # it's a length
```

Comment:
```suggestion
            if type(colType) is int: # it's a length
```
`type(0)` seems kinda weird, it's always returns a `int`.

Reply:
But why change the isinstance check at all?